src/Wizard.jl

```diff
@@ -1145,25 +1139,353 @@ function step7(state::WizardState)
       email: false
     git:
       depth: 99999999
-    sudo: required
-    services:
-      - docker
-
+    cache:
+      timeout: 1000
+      directories:
+        - downloads
     env:
-      DOCKER_IMAGE: staticfloat/julia_crossbuild:x64
+      global:
+        - BINARYBUILDER_DOWNLOADS_CACHE=downloads
+    sudo: required
 
     # Before anything else, get the latest versions of things
     before_script:
-      - docker pull \$DOCKER_IMAGE
       - julia -e 'Pkg.clone("https://github.com/JuliaPackaging/BinaryProvider.jl")'
       - julia -e 'Pkg.clone("https://github.com/JuliaPackaging/BinaryBuilder.jl"); Pkg.build()'
 
     script:
       - julia build_tarballs.jl
-    ```
+      
+    before_cache:
+      - rm -rf \$(julia -e 'println(Pkg.dir("BinaryBuilder","deps","root"))')
+    """)
+end
+
+function print_travis_deploy(io, repo, secure_key)
+    print(io,
+    """
+    deploy:
+        provider: releases
+        api_key:
+            # Note; this api_key is only valid for $(GitHub.name(repo)); you need
+            # to make your own: https://docs.travis-ci.com/user/deployment/releases/
+            secure: $secure_key
+        file_glob: true
+        file: products/*.tar.gz
+        skip_cleanup: true
+        on:
+            repo: $(GitHub.name(repo))
+            tags: true
     """)
 end
 
+mutable struct termios
+	c_iflag::Cint
+	c_oflag::Cint
+	c_cflag::Cint
+	c_lflag::Cint
+	c_line::UInt8
+	c_cc::NTuple{19, UInt8}
+    termios() = new()
+end
+
+const TCGETS = 0x5401
+const TCSETS = 0x5402
+const ECHO = 0o10
+
+function set_terminal_echo(fd, enable::Bool)
+    # Linux only
+    t = termios()
+    ccall(:ioctl, Cint, (Cint, Cint, Ref{termios}), fd, TCGETS, t)
+    old = (t.c_lflag & ECHO) != 0
+    if enable
+        t.c_lflag |= ECHO
+    else
+        t.c_lflag &= ~ECHO
+    end
+    ccall(:ioctl, Cint, (Cint, Cint, Ref{termios}), fd, TCSETS, t)
+    old
+end
+
+const travis_headers = Dict(
+    "Accept" => "application/vnd.travis-ci.2+json",
+    "Content-Type" => "application/json",
+    "User-Agent" => "Travis-WHY-YOU-HAVE-BAD-DOCS (https://github.com/travis-ci/travis-ci/issues/5649)"
+)
+
+function authenticate_travis(github_token)
```

Comment:
Do we know if 2FA screws this up at all?

Reply:
No, 2FA works fine. I have it enabled.